knox.go

```diff
@@ -38,10 +40,16 @@ func NewKnoxClient(c *KnoxConfig) (*KnoxClient, error) {
 		return &KnoxClient{}, err
 	}
 
+	tokenClient, err := token.NewTokenClient(c.Network.AuthServiceURL, c.Signer)
+	if err != nil {
+		return &KnoxClient{}, err
```

Comment:
`return nil, err`

Reply:
I don't think we should return nil for user facing SDKs, I'd prefer to prevent those kinds of nil panics. I'm fine with returning nil for internal usage but not external user-facing stuff

Reply:
I just don't think the memory differences are meaningful